src/aktualizr_secondary/main.cc

```diff
@@ -85,10 +88,18 @@ int main(int argc, char *argv[]) {
   int ret = EXIT_SUCCESS;
   try {
     AktualizrSecondaryConfig config(commandline_map);
-    LOG_DEBUG << "Current directory: " << boost::filesystem::current_path().string();
+    AktualizrSecondary::Ptr secondary;
 
-    auto secondary = AktualizrSecondaryFactory::create(config);
-    SecondaryTcpServer tcp_server(secondary->getDispatcher(), config.network.primary_ip, config.network.primary_port,
+    if (config.pacman.type != PACKAGE_MANAGER_OSTREE) {
+      secondary = std::make_shared<AktualizrSecondaryFile>(config);
+    }
+#ifdef BUILD_OSTREE
+    else {
+      secondary = std::make_shared<AktualizrSecondaryOstree>(config);
+    }
+#endif  // BUILD_OSTREE
```

Comment:
> I think we should probably report at error if you try to use the OSTree package manager but didn't enable OSTree.

How exactly to report it and to what entity? There is no means to report errors from Secondary to BE at the moment. If you mean just outputting log message then I can added it but it will be output anyway in the exception catch statement.

Reply:
The log is good enough for now.

Reply:
done